src/App.js

```diff
@@ -1,25 +1,75 @@
-import logo from './logo.svg';
-import './App.css';
+import { useEffect, useState } from "react"
+import Confetti from "react-confetti"
+import Die from "./Die"
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+export default function App() {
+    const [dice, setDice] = useState(allNewDice())
+    const [tenzies, setTenzies] = useState(false)
+    
+    useEffect(() => {
+        const firstValue = dice[0].value
+        const allHeld = dice.every(die => die.held)
+        const allSameNumber = dice.every(die => die.value === firstValue)
+        if(allHeld && allSameNumber) {
+            setTenzies(true)
+        }
+    }, [dice])
+
+    const styles = {
+      width : tenzies ? "150px" : "100px"
+    }
+    
+    function randomDieValue() {
+        return Math.ceil(Math.random() * 6)
+    }
+
+    function allNewDice() {
+        const newArray = []
+        for(let i = 0; i < 10; i++) {
+            const newDie = {
+                value: randomDieValue(),
+                held: false,
+                id: i + 1
+            }
+            newArray.push(newDie)
+        }
+        return newArray
+    }
+
+    function rollUnheldDice() {
+        if (!tenzies) {
+            setDice((oldDice) => oldDice.map((die, i) =>
+                die.held ? 
+                    die : 
+                    { value: randomDieValue(), held: false, id: i + 1 }
+            ))
+        } else {
+            setDice(allNewDice())
+            setTenzies(false)
+        }
+    }
 
-export default App;
+    function holdDice(id) {
+        setDice(prevDice => prevDice.map(die => {
+            return die.id === id ? 
+                {...die, held: !die.held} : 
+                die
+        }))
+    }
+
+    const diceElements = dice.map((die) => (
```

Comment:
No need to wrap the die inside the brackets, as it only one parameter you can leave the brackets... there's no need of them.unless you have multiple parameters